### DISCUSSION TITLE
Cannot connect the HA integration to the dongle via modbus - logs included

### DISCUSSION THREAD

**Discussion by digijoebz** [wlcrs/huawei_solar#772] [Setting up the integration]
Hi - I am struggling to connect HA to my inverter's dongle. I have enabled modbus. 
I tried to connect to both ports, logs are shown below.  My installation seems fairly standard, I have 2 inverters + dongle.  The only thing that is different is that the dongle is on a different subnet (192.168.1.x) to my HA installation (192.168.0.1). However I can ping the dongle from HA's terminal.  I've also included a screenshot of the inverter firmware 
![IMG_7693](https://github.com/user-attachments/assets/82a2f254-6381-4b2b-9167-226b357f7d69)

and the dongle in the dhcp pool
<img width="871" alt="Screenshot 2024-07-17 at 10 07 41" src="https://github.com/user-attachments/assets/820a859d-0009-4724-9f7e-fcf163c701d9">

A bit stuck what to try next, any help appreciated! 
thanks 
Joe

 via 501

----------------------------------------------------------------

Logger: huawei_solar.huawei_solar
Source: /usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py:203
First occurred: 9:49:43 AM (1 occurrences)
Last logged: 9:49:43 AM

Aborting client creation due to error.
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/locks.py", line 212, in wait
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 199, in create
    await huawei_solar._initialize()
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 118, in _initialize
    self.time_zone = (await self.get(rn.TIME_ZONE)).value
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 271, in get
    return (await self.get_multiple([name], slave))[0]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 308, in get_multiple
    response = await self._read_registers(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 444, in _read_registers
    async with self._communication_lock():
  File "/usr/local/lib/python3.12/contextlib.py", line 210, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 157, in _communication_lock
    await asyncio.wait_for(
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 519, in wait_for
    async with timeouts.timeout(timeout):
  File "/usr/local/lib/python3.12/asyncio/timeouts.py", line 115, in __aexit__
    raise TimeoutError from exc_val
TimeoutError

----------------------------------------------------------------

via the 6607

Logger: pymodbus.logging
Source: runner.py:190
First occurred: 9:49:43 AM (1 occurrences)
Last logged: 9:49:43 AM

Failed to connect [Errno 111] Connect call failed ('192.168.1.10', 6607)

**Comment by Roving-Ronin** [comment]
@digijoebz 
You've followed the info at https://github.com/wlcrs/huawei_solar/wiki/Connecting-to-the-inverter#network-connection for connecting to the sDongle, that is enabled for modbus access and set to unlimited (it has options to limit the source IP for connections).

The different subnet, unless your doing filtering between subnet/VLANs, should not pose a problem. In my own setup with 2 x M1 6kW cascaded and a sDongle in the primary inverter they are also on different subnets and VLANs.

You might want to login to FusionSolar --> Device Management and run an update on the Modus version. You have SPC320 and SPC330 was released near 2 years ago (there is a newer version again...SPC335 iirc, but depending upon where you live, Huawei support may not make it available to you).

**Comment by digijoebz** [comment]
Thanks @Roving-Ronin for the quick reply!  
Yes Modbus was set to unlimited. I think I know the problem, as per reading the [Prerequisites](https://github.com/wlcrs/huawei_solar?tab=readme-ov-file#prerequisites) wiki, I thought I needed to upgrade the **inverter** firmware to v200, I upgraded the inverter and the modbus fw to the latest, but it seems the sdongle is the one that goes to v200.  Now have to figure out how to reset the sdongle as the installer changed the password.

**Comment by digijoebz** [comment]
Btw the inverter firmware they gave me was the SUN2000MAV100R001C00SPC164. They said V200 does not exist for the M1 - I have a SUN2000-6KTL-M1

**Reply by Roving-Ronin** [reply]
Use SPC160 on M1, v2xxx..SPC110 on sDongle, MBUS 330.
